scripts/create-preview-tabs.ts

```diff
@@ -46,3 +47,30 @@ export function createPreviewTabs(props: Props) {
 
   return tabs;
 }
+
+const joinStrs = (strs: string[]) => {
+  return `[${strs.map(str => `"${str}"`).join(", ")}]`;
+};
+
+export const CreateToastSandbox = ({
+  type = "tsx",
+  deps = [],
+}: {
+  type?: string;
+  deps?: string[];
+}) => {
+  const utilTab = type === "tsx" ? "Utils.tsx" : "Utils.jsx";
+  const ReactTab = type === "tsx" ? "React" : "ReactJS";
+
+  return new Function(`
+  var previews = arguments[0];
+  return {
+      framework: "${ReactTab.toLowerCase()}",
+      files: {
+          "src/App.${type}": previews["${ReactTab}"][0],
+          "src/styles.css": previews["CSS"] ? previews["CSS"][0] : "",
+          "src/ToastUtils.component.${type}": previews["${utilTab}"] ? previews["${utilTab}"][0] : "",
+      },
+      userDependencies: ${joinStrs(deps)},
+  };`) as CodeSandboxTemplate;
+};
```

Comment:
Tried using normal higher order function but does not seem to be working. throwing `type` not defined error. i think internally we loose the function context and i think thats why addon-preview also uses string template literal functions

Reply:
Oh okay okay !!